### PR TITLE
METRON-838 Incorrect set of ts in FireEye parser

### PR DESCRIPTION
**Pull request by bjigmp**
## Contributor Comments
Although log line is parsed (method getTimeStamp of BasicFireEyeParser class, line 120) and day/month/year are extracted ts is not set to correct value. It set in line 128 but with default null month, day and time

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron (Incubating).  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [ ] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root incubating-metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  bin/generate-md.sh
  mvn site:site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommened that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by ottobackwards** [comment]
Hi bjigmp!
Thanks for the contribution.
We have a template for our PR's, that help us review and accept them.  This PR is missing that,
can you please add it in?

**Comment by cestella** [comment]
Thanks for the contribution!  Also, could you please close and reopen the PR?  You hit an intermittent test failure and travis should try again.

**Comment by ottobackwards** [comment]
I created a gist of the template that you can copy and paste.
Please go through and fill out the appropriate things.

https://gist.github.com/ottobackwards/9b7ea0689a79b9510c52ad49045aea7e

**Comment by bjigmp** [comment]
Thanks @ottobackwards, Updated this PR. Are all checkboxes must be checked?

**Comment by ottobackwards** [comment]
No just the appropriate ones you can see.

**Comment by ottobackwards** [comment]
Hi, What is the status of this PR?

**Comment by bjigmp** [comment]
Hi Otto, sorry for delay. I did not have sample data so I was not able to provide reproducible steps or write unit/integration tests. But recently I found some sampledata for FireEye in metron-platform/metron-parsers/src/test/resources/logData/FireEyeParserTest.txt
So I'm going to write integration test and add to this PR

**Comment by bjigmp** [comment]
Wrote test for FireEye and found that it uses ParserUtils.convertToEpoch that returns incorrect value.
Filed METRON-1003. Will create PR to fix this bug and then continue with this PR.

**Comment by justinleet** [comment]
@bjigmp I just merged in https://github.com/apache/metron/pull/623, so you should be able to continue this now. Thanks again for the submissions!

**Comment by justinleet** [comment]
@bjigmp When you have a chance, could you deconflict this?

File: metron-platform/metron-parsers/src/test/java/org/apache/metron/parsers/fireeye/BasicFireEyeParserTest.java [hunk 1]
```diff
@@ -57,4 +62,16 @@ public void testParse() throws ParseException {
...
+
+  private final static String fireeyeMessage = "<164>Mar 19 05:24:39 10.220.15.15 fenotify-851983.alert: CEF:0|FireEye|CMS|7.2.1.244420|DM|domain-match|1|rt=Feb 09 2015 12:28:26 UTC dvc=10.201.78.57 cn3Label=cncPort cn3=53 cn2Label=sid cn2=80494706 shost=dev001srv02.example.com proto=udp cs5Label=cncHost cs5=mfdclk001.org dvchost=DEVFEYE1 spt=54527 dvc=10.100.25.16 smac=00:00:0c:07:ac:00 cn1Label=vlan cn1=0 externalId=851983 cs4Label=link cs4=https://DEVCMS01.example.com/event_stream/events_for_bot?ev_id\\=851983 dmac=00:1d:a2:af:32:a1 cs1Label=sname cs1=Trojan.Generic.DNS";
+
+  @SuppressWarnings("rawtypes")
+  @Test
+  public void testTimestampParsing() throws ParseException {
+    JSONObject parsed = parser.parse(fireeyeMessage.getBytes()).get(0);
+    JSONParser parser = new JSONParser();
+    Map json = (Map) parser.parse(parsed.toJSONString());
+    long expectedTimestamp = ZonedDateTime.of(Year.now(UTC).getValue(), 3, 19, 5, 24, 39, 0, UTC).toInstant().toEpochMilli();
```
Comment (justinleet) [inline metron-platform/metron-parsers/src/test/java/org/apache/metron/parsers/fireeye/BasicFireEyeParserTest.java]: It's incredibly minor (and optional), but we could just swap out the `ZoneId.of("UTC")` for `ZoneOffset.UTC`

At that point, this changes slightly, but still seems reasonable
```
    long expectedTimestamp = ZonedDateTime.of(
        Year.now(ZoneOffset.UTC).getValue(),
        3,
        19,
        5,
        24,
        39,
        0,
        ZoneOffset.UTC
    ).toInstant().toEpochMilli();
```

**Comment by justinleet** [comment]
+1 to this.  There's a review comment, but quite frankly I consider it optional.  If @bjigmp doesn't want to make that change, I'm still good.

**Comment by bjigmp** [comment]
Thanks @justinleet. Changed to ZoneOffset.UTC

**Pull request closed without merging**